### PR TITLE
vim-patch:8.1.{2185,2188,2190,2197}

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

**Comment by blueyed** [comment]
FreeBSD was/is flaky, restarted Travis.

**Comment by justinmk** [comment]
`ci/install.sh` failing, did treesitter upstream files change? are we not pinned to a specific treesitter version?

```
No such file or directory (os error 2)
The command "ci/install.sh" failed and exited with 1 during .
```

**Comment by blueyed** [comment]
> `ci/install.sh` failing, did treesitter upstream files change? are we not pinned to a specific treesitter version?
> 
> ```
> No such file or directory (os error 2)
> The command "ci/install.sh" failed and exited with 1 during .
> ```

```
> tree-sitter-cli@0.15.11 install /home/travis/.nvm/versions/node/v8.12.0/lib/node_modules/tree-sitter-cli
> node install.js
Downloading https://github.com/tree-sitter/tree-sitter/releases/download/0.15.11/tree-sitter-linux-x64.gz
+ tree-sitter-cli@0.15.11
added 1 package from 1 contributor in 1.479s
/home/travis/build/neovim/neovim/node_modules/tree-sitter-cli -> /home/travis/.nvm/versions/node/v8.12.0/lib/node_modules/tree-sitter-cli
Install tree-sitter c parser
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  252k    0  252k    0     0   526k      0 --:--:-- --:--:-- --:--:--  526k
ambiguities:
  ✓ pointer declarations vs multiplications
  ✓ casts vs multiplications
  ✓ function-like type macros vs function calls
  ✓ function calls vs parenthesized declarators vs macro types
  ✓ Call expressions vs empty declarations w/ macros as types
  ✓ Comments after for loops with ambiguities
  ✓ Top-level macro invocations
types:
  ✓ Primitive types
  ✓ Type modifiers
crlf:
  ✓ Line comments with escaped CRLF line endings
preprocessor:
  ✓ Include directives
  ✓ Object-like macro definitions
  ✓ Function-like macro definitions
  ✓ Ifdefs
  ✓ General if blocks
  ✓ Preprocessor conditionals in functions
  ✓ Preprocessor conditionals in struct/union bodies
  ✓ Unknown preprocessor directives
declarations:
  ✓ Struct declarations
  ✓ Union declarations
  ✓ Enum declarations
  ✓ Struct declarations containing preprocessor directives
  ✓ Primitive-typed variable declarations
  ✓ Variable storage classes
  ✓ Composite-typed variable declarations
  ✓ Pointer variable declarations
  ✓ Typedefs
  ✓ Function declarations
  ✓ Function definitions
  ✓ Function specifiers after types
  ✓ Linkage specifications
  ✓ Type qualifiers
  ✓ Local array declarations
  ✓ Attributes
expressions:
  ✓ Number literals
  ��� Identifiers
  ✓ Common constants
  ✓ Function calls
  ✓ String literals
  ✓ Character literals
  ✓ Field access
  ✓ Boolean operators
  ✓ Math operators
  ✓ The comma operator
  ✓ Assignments
  ✓ Pointer operations
  ✓ Type-casts
  ✓ Sizeof expressions
  ✓ Compound literals
  ✓ Compound literals with trailing commas
  ✓ Comments with escaped newlines
  ✓ Comments with escaped chars and newlines
statements:
  ✓ If statements
  ✓ For loops
  ✓ While loops
  ✓ Labeled statements
  ✓ Switch statements
  ✓ Comments with asterisks
No such file or directory (os error 2)
```
/cc @bfredl

**Comment by janlazo** [comment]
nvm `--lts` is aliased to v8.12.x. v8.16.x is EOL by end of this year. tree-sitter runs on Node v10.x (LTS) (https://github.com/tree-sitter/tree-sitter/commit/ad43b211f4930404563a3ec65ba1a75e0948c9fc#diff-354f30a63fb0907d4ad57269548329e3R14-R15).

**Comment by blueyed** [comment]
See also https://github.com/neovim/neovim/pull/11266 wrt tree-sitter.

File: ci/build.ps1 [hunk 1]
```diff
@@ -98,27 +98,17 @@ npm.cmd install -g neovim
...
-
-$client.DownloadFile("https://codeload.github.com/tree-sitter/tree-sitter-c/zip/v0.15.2","c:\treesitter\tree_sitter_c.zip")
-Expand-Archive c:\treesitter\tree_sitter_c.zip -DestinationPath c:\treesitter\
-cd c:\treesitter\tree-sitter-c-0.15.2
-c:\treesitter\tree-sitter.exe test
+$client.DownloadFile("https://codeload.github.com/tree-sitter/tree-sitter-c/zip/v0.15.2",".\tree_sitter_c.zip")
+Expand-Archive .\tree_sitter_c.zip -DestinationPath .
+cd tree-sitter-c-0.15.2
+tree-sitter.cmd test
+Test-Path -PathType Leaf "$TREE_SITTER_DIR\bin\c.dll"
```
Comment (janlazo) [inline ci/build.ps1]: Why is the functional test failing when this test (for the same path) succeeds?

**Pull request merged**